aws/data_source_aws_vpcs_test.go

```diff
@@ -0,0 +1,149 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccDataSourceAwsVpcs_basic(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceAwsVpcsConfig(),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsVpcsDataSourceID("data.aws_vpcs.all"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccDataSourceAwsVpcs_tags(t *testing.T) {
+	rName := acctest.RandString(5)
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceAwsVpcsConfig_tags(rName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("data.aws_vpcs.selected", "ids.#", "1"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccDataSourceAwsVpcs_filters(t *testing.T) {
+	rName := acctest.RandString(5)
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceAwsVpcsConfig_filters(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckResourceAttrGreaterThanValue("data.aws_vpcs.selected", "ids.#", "0"),
+				),
+			},
+		},
+	})
+}
+
+func testCheckResourceAttrGreaterThanValue(name, key, value string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		ms := s.RootModule()
+		rs, ok := ms.Resources[name]
+		if !ok {
+			return fmt.Errorf("Not found: %s in %s", name, ms.Path)
+		}
+
+		is := rs.Primary
+		if is == nil {
+			return fmt.Errorf("No primary instance: %s in %s", name, ms.Path)
+		}
+
+		if v, ok := is.Attributes[key]; !ok || !(v > value) {
+			if !ok {
+				return fmt.Errorf("%s: Attribute '%s' not found", name, key)
+			}
+
+			return fmt.Errorf(
+				"%s: Attribute '%s' is not greater than %#v, got %#v",
+				name,
+				key,
+				value,
+				v)
+		}
+		return nil
+
+	}
+}
+
+func testAccCheckAwsVpcsDataSourceID(n string) resource.TestCheckFunc {
```

Comment:
Minor nitpick: We will have the documentation up very shortly (https://github.com/hashicorp/terraform-website/pull/317 😉 ), but even when we're not checking for resource/datasource API object, we prefer to name these functions `testAccCheckXXXExists` for consistency. I'll adjust this on merge.

Reply:
Sure 👍